### PR TITLE
E2115 choose mentor

### PR DESCRIPTION
**Pull request by AvlWx2014**
* Add: new constant `Participant#DUTY_MENTOR` representing the duty title of mentor for the database. This will identify participants as mentors.
* Add: `MentorManagement` class, and class methods for the select mentor functionality.

### CODE REVIEW

**Review by bahati2000 — COMMENTED** [review]
I wonder what would be the trigger for this class. Because mentors will be assigned when an assignment without topics is created. Should the code to check when this is created be included in this class?  On second thought I don't think so. This might violate single responsibility principle.

**Review by bahati2000 — APPROVED** [review]

**Review by bahati2000 — APPROVED** [review]

**Review by bahati2000 — APPROVED** [review]

**Review by bahati2000 — APPROVED** [review]

**Review by bahati2000 — APPROVED** [review]

**Review by bahati2000 — APPROVED** [review]

**Review by bahati2000 — APPROVED** [review]

**Comment by molinamelendezj** [comment]
@bahati2000 really wants these changes in beta lol getting blasted by approved emails 😆

File: app/models/mentor_management.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  #
+  # @return The id of the mentor with the fewest teams
+  #   they are assigned to. Returns `nil` if there are
+  #   no participants with mentor duty for [assignment_id].
+  def self.select_mentor(assignment_id)
+    mentor_user_id, _ = self.zip_mentors_with_team_count(assignment_id).first
+    User.where(id: mentor_user_id).first
+  end
+
+  def self.update_mentor_state(assignment_id, team_id)
```
Comment (molinamelendezj) [inline app/models/mentor_management.rb:43]: considering rename to `assign_mentor()`?? add comments of what this method does.

File: app/models/mentor_management.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  def self.get_all_mentors
+    Participant.where(duty: Participant::DUTY_MENTOR)
+  end
+
+  # Select all the participants who's duty in the participant
+  # table is [DUTY_MENTOR], and who are a participant of
+  # [assignment_id].
+  #
+  # @see participant.rb for the value of DUTY_MENTOR
+  def self.get_mentors_for_assignment(assignment_id)
```
Comment (molinamelendezj) [inline app/models/mentor_management.rb:103]: consider rename

File: app/models/invitation.rb [hunk 3]
```diff
@@ -61,7 +61,8 @@ def self.accept_invite(team_id, inviter_user_id, invited_user_id, assignment_id)
 
     if can_add_member # The member was successfully added to the team (the team was not full)
       Invitation.update_users_topic_after_invite_accept(inviter_user_id, invited_user_id, assignment_id)
-
+      new_team_id = TeamsUser.team_id(assignment_id, inviter_user_id)
```
Comment (molinamelendezj) [inline app/models/invitation.rb:64]: consider create issue?

File: app/models/invitation.rb [hunk 4]
```diff
@@ -61,7 +61,8 @@ def self.accept_invite(team_id, inviter_user_id, invited_user_id, assignment_id)
...
+      MentorManagement.update_mentor_state(assignment_id, new_team_id)
```
Comment (AvlWx2014) [inline app/models/invitation.rb:65]: Add a comment for the presence of this code here.

File: app/models/teams_user.rb [hunk 5]
```diff
@@ -6,7 +6,11 @@ class TeamsUser < ActiveRecord::Base
   attr_accessible :user_id, :team_id
 
   def name(ip_address = nil)
-    self.user.name(ip_address)
+    name = self.user.name(ip_address)
+    if MentorManagement.user_a_mentor?(self.user)
```
Comment (AvlWx2014) [inline app/models/teams_user.rb:10]: Leave a comment for the reason this is here.

**Comment by AvlWx2014** [comment]
Merging so we can get this in to beta.

**Pull request merged**